frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -67,7 +67,7 @@ function ReaderPaging:registerKeyEvents()
         self.key_events.GotoNextPos = { { "Down" }, event = "GotoPosRel", args = 1, }
         self.key_events.GotoPrevPos = { { "Up" }, event = "GotoPosRel", args = -1, }
     end
-    if Device:hasKeyboard() then
+    if Device:hasKeyboard() and not Device.alt_plus_key_kernel_translated then
```

Comment:
This seems unidiomatic. Shouldn't it be something like `Device:getKernelKeyTranslationMap()`?

Reply:
Yes, it is unidiomatic - but making it idomatic and super generic (with :hasKernelKeyTranslationMap and :get.. - without the "Alt" in the name, meaning it could/should handle every case ) would need more thinking - and we have only one instance (Kindle3) where this would be used.
So, I'd be ok with this quick hack, it just needs to read well and natural :)

Reply:
In that case it should be something like k3_ or kindle3_ followed by… hm, alt_modified_keys_kernel_translated? alt_keys_kernel_translated?

Reply:
Dunno. Not even sure about my suggestion now :)
May be all Device should have a `getKernelTranslated(cod) return cod end`, and Kindle3 would really have some implementation, and we should use (everywhere, or juste where we know it's needed):
```lua
    if Device:hasKeyboard() then
        self.key_events.ShowWikipediaLookup = { Device.getKernelTranslated({ "Alt", "W" }), { "Ctrl", "W" } }
```
Let's wait for @NiLuJe thoughts.

Reply:
Make up your minds fellow humans ;) i thought it needed to be in something_something format @poire-z? What is this blatant disregard for the rules and order? 😄

Reply:
`variable_name`, `functionOrMethodName()`, `ClassName`.

`AltPlusTopRowKeys` as a name limits its meaning/usability in other contexts (even if there is no other at the moment and probably never).

Reply:
> i thought it needed to be in something_something format

That's somewhat tangential. We use aFunction and a_variable so you don't have to guess. So the writing style wasn't right for what it was but that ignores the deeper question as to whether ot should be that in the first place. ^_^

Reply:
To be or not to be, note: great idea for a soliloquy. Someone should run with it.

Reply:
> Let's wait for @NiLuJe thoughts.

Yep, I'd stick with a K3-specific approach, and just check for the existence of the table, with a k3_ prefix to make that doubly clear.

I don't expect we'll ever see another device with a built-in keyboard ;).

----

Also, mildly related, but I forgot to mention it the last time: I'd drop the `+` from the filename, as I wouldn't trust FAT32/vfat/fuse to not fuck that one up.